feat: docs for b2b sso (#1544)
ory/docs: docs/kratos/organizations/organizations.mdx

---
id: organizations
title: Organizations and B2B Single Sign-On
---

```mdx-code-block
import Mermaid from "@site/src/theme/Mermaid"
```

> ## Introduction

Reviewer: ```suggestion
```

IMO the introductory text is enough; and best practice is not to put headlines after each other with no text between.

Organizations are a way to group users together within one project. An organization always belongs to one Ory project. Within a
project, an identity can belong to an organization or remain without an organization. Organizations contain multiple OIDC SSO
connections. All members of an organizations must use one of the organization's OIDC SSO connections to log in.

An organization can have multiple domains. Registrations for email addresses with a domain that belongs to an organization must go
through one of the organization's OIDC SSO connections.

```mdx-code-block
<Mermaid
  chart={`
graph LR
    Project -->|has many| Organization
    Organization -->|has many| Users
    Organization -->|has many| Domain
    Organization -->|has many| Connection
    Users -.->|authenticate| Connection
    Project -.->|Identity pool| Users
    Connection -.- Google
    Connection -.- Okta
    Connection -.- Keycloak
`}
/>
```

> ## Managing organizations

Reviewer: ```suggestion
## Manage organizations
```

the present form reads better than gerund from in headlines in general IMO

#     "label": "My Organization",
#     "domains": [
#       "my.example.com"
#     ]
#   }
```

### List the project's organizations

```shell
# project_id = your project ID
# ory_session_cookie = the value of the ory_session_ory cookie

curl -X GET --location "https://api.console.ory.sh/projects/${project_id}/organizations" \
    -H "Content-Type: application/json" \
    -H "Cookie: ory_session_ory=${ory_session_cookie}"

# Returns:
#   {
#     "organizations": [
#       {
#         "id": "696778f4-8975-45ad-8809-5433dc0f576e",
#         "project_id": "45a22e6f-5eca-4925-bc44-4736c775e892",
#         "label": "My Organization",
#         "domains": [
#           "my.example.com"
#         ]
#       }
#     ]
#   }
```

### Delete an organization

```shell
# organization_id = your organization ID (from a previous GET request)
# project_id = your project ID
# ory_session_cookie = the value of the ory_session_ory cookie

curl -X DELETE --location "https://api.console.ory.sh/projects/${project_id}/organizations/${organization_id}" \
    -H "Content-Type: application/json" \
    -H "Cookie: ory_session_ory=${ory_session_cookie}"

# Returns an empty response on success
```

> ## Creating SSO connections for an organization

Reviewer: ```suggestion
## Create SSO connections for an organization
```

Same as above, to keep all headlines uniform

After creating an organization, continue by adding one or more SSO OIDC connections. In the Ory Console, navigate to
[Authentication > Social Sign-In & SSO (OIDC)](https://console.ory.sh/projects/current/social-signin). From there, add a new

> OpenID Connect provider. Under "show advanced settings", configure the organization to use for this connection.

Reviewer: ```suggestion
OpenID Connect provider. Under **Show advanced settings**, configure the organization to use for this connection.
```

[Bold UI labels](https://github.com/ory/docs#ui-references)

![Configure the SSO connection in the Ory Console](configure-sso-connection.png)

> Next, go to your registration page. Entering an email that ends with the organization's domain (`@my.example.com` in the example
> above) shows a "Sign in with SSO" button instead of the passowrd field. Clicking it will take you to sign in with the SSO

Reviewer: ```suggestion
Next, go to your registration page. Entering an email that ends with the organization's domain, such as `@my.example.com` from the example
above, shows a **Sign in with SSO** button instead of the password field. Clicking it will take you to sign in with the SSO
```
No parantheses please

> connection. Also note that the SSO connection is not visible for email addresses that are not managed by the organization.

Reviewer: ```suggestion
connection. The SSO connection is not visible for email addresses that are not managed by the organization.
```
